behavior_tests/src/user-defined-sycl-file-extension/do_test.py

```diff
@@ -0,0 +1,67 @@
+# ====------ do_test.py---------- *- Python -* ----------------------------===#
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# ===----------------------------------------------------------------------===#
+import os
+import tempfile
+
+from test_utils import *
+from test_config import CT_TOOL
+
+
+def setup_test():
+    change_dir(test_config.current_test)
+    return True
+
+
+def migrate_test():
+    """
+    Runs dpct with options for --sycl-file-extension and verfies the exisitance
+    of migrated files with correct extension
+    """
+    sub_command = f"{test_config.CT_TOOL} main.cu --out-root {{0}}"
+
+    # commands to test three options to --sycl-file-extension and the defualt
+    # behaviour
+    cmds_n_results = {
+        f"{sub_command} --sycl-file-extension=sycl-cpp": "main.sycl.cpp",
+        f"{sub_command} --sycl-file-extension=dp-cpp": "main.dp.cpp",
+        f"{sub_command} --sycl-file-extension=cpp": "main.cpp",
```

Comment:
Pls add source with extension name ".cpp", for example foo.cpp
Then what extension name is used for foo.cpp when each of "--sycl-file-extension=sycl-cpp, --sycl-file-extension=dp-cpp, and --sycl-file-extension=cpp" is specified 
Also, what the default extension name is used for main.cu, foo.cpp when all these three extension option are not specified.

Reply:
Added _src.cpp_, _src.h_, _cuda_src.cuh_.

The _.cpp_ and _.h_ remains same in all the cases supported by the option _-sycl-file-extension_.

The default extension name is _.dp.cpp_ and _dp.hpp_. There is already a test related to it.